homeassistant/components/planifneige/__init__.py

```diff
@@ -0,0 +1,100 @@
+"""Platform for the City of Montreal's Planif-Neige snow removal APIs."""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.const import CONF_API_KEY, CONF_NAME, CONF_SCAN_INTERVAL
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+
+DOMAIN = 'planifneige'
+DATA_PLANIFNEIGE = 'data_planifneige'
+
+DATA_UPDATED = '{}_data_updated'.format(DOMAIN)
+
+PLANIFNEIGE_ATTRIBUTION = "Information provided by the City of Montreal "
+
+REQUIREMENTS = ['planif-neige-client==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_STREETID = 'street_id'
+CONF_STREETS = 'streets'
+
+DEFAULT_INTERVAL = timedelta(minutes=5)
+
+_STREET_SCHEME = vol.Schema({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_STREETID): cv.positive_int
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Optional(CONF_SCAN_INTERVAL,
+                     default=DEFAULT_INTERVAL): vol.All(
+                         cv.time_period, cv.positive_timedelta),
+        vol.Required(CONF_STREETS): [_STREET_SCHEME]
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Set up the PlanifNeige component."""
+    from planif_neige_client import planif_neige_client
+    db_path = hass.config.path('planifneige.db')
+    conf = config[DOMAIN]
+
+    pn_client = planif_neige_client.PlanifNeigeClient(conf.get(
```

Comment:
Use `dict[key]` for required config keys and keys with default config schema values.